stringtemplate4/src/main/java/org/jdbi/v3/stringtemplate4/StringTemplateSqlLocator.java

```diff
@@ -130,7 +131,13 @@ public static STGroup findStringTemplateGroup(String path) {
      * @return the loaded StringTemplateGroup.
      */
     public static STGroup findStringTemplateGroup(ClassLoader classLoader, String path) {
-        return CACHE.computeIfAbsent(path, p -> readStringTemplateGroup(classLoader, path));
+        final STGroup cached = CACHE.get(path);
+        if (cached != null) {
+            return cached;
+        }
+        synchronized (StringTemplateSqlLocator.class) {
```

Comment:
Let's synchronize on a private object rather than the class to be safe.

Reply:
#1090 i've used a privat static final Object in my PR.